src/dodal/devices/i18/diode.py

```diff
@@ -11,6 +38,8 @@ def __init__(
         name: str = "",
     ):
         self._prefix = prefix
+        self.positioner_a = create_positioner(FilterAValues, prefix + ":A:MP")
+        self.positioner_b = create_positioner(FilterBValues, prefix + ":B:MP")
```

Comment:
Should: These almost certainly want to be read, so should be inside the add_children_as_readables
```suggestion
        self.positioner_a = create_positioner(FilterAValues, prefix + "A")
        self.positioner_b = create_positioner(FilterBValues, prefix + "B")
```

Also this was found to connect to stage_position but other problems with _stage_motion aside.